### ISSUE TITLE
Cannot set null to DateTimeColumn

### ISSUE THREAD

**Issue by DennisMuchiri** [simolus3/moor#795]
I am trying to update my object like this

 widget.erpticket = widget.erpticket.copyWith(
                    close_date: null,
                    issettobeupdated: true,
                  );

But when I log

print(widget.erpticket.close_date);

The value is what I had set before

2020-09-01 12:55:39.424250

Yet close_date is nullable

DateTimeColumn get close_date => dateTime().nullable()();

What am I doing wrong? Is this a bug?

**Comment by DennisMuchiri** [comment]
@simolus3 pleae advise

**Comment by simolus3** [comment]
This is working as intended, but was a poor design choice I made...
The default value of the `close_date` parameter in `copyWith` is null. When receiving a `null` value, `copyWith` will assume that the parameter wasn't set and use the old value. You can probably get around this now by converting the `erpticket` to a companion, updating that, and then converting that back to a row (using [`mapFromCompanion`](https://pub.dev/documentation/moor/latest/moor/TableInfo/mapFromCompanion.html)):

```dart
final companion = widget.erpticket.toCompanion(true).copyWith(
  close_date: Value(null),
  issettobeupdated: Value(true),
);
widget.erpticket = database.erptickets.mapFromCompanion(companion);
```
 
I'll work on a build option to generate `Value<...>` parameters in `copyWith` when the underlying column is nullable. That should fix this problem.

**Comment by DennisMuchiri** [comment]
@simolus3 thank you let me try this

**Comment by DennisMuchiri** [comment]
@simolus3 I am getting an error

E/flutter ( 6497): [ERROR:flutter/lib/ui/ui_dart_state.cc(157)] Unhandled Exception: type 'DateTime' is not a subtype of type 'int' in type

**Comment by simolus3** [comment]
Can you post the stack trace of that exception?

**Comment by DennisMuchiri** [comment]
@simolus3  here
![stckov11111](https://user-images.githubusercontent.com/11217100/91849931-76f76c00-ec65-11ea-855c-a01fe2821b09.PNG)

**Comment by simolus3** [comment]
Thanks, I think I know where this is coming from now. This is probably because of https://github.com/simolus3/moor/issues/559, which should have been fixed in version `3.1.0`. Can you try upgrading if you're using an older version? The latest moor version is `3.3.1`.

**Comment by DennisMuchiri** [comment]
Yes I was using version 2.3.1. This has worked. Thank you @simolus3 .Closing
